### PR TITLE
Functions as observers

### PR DESCRIPTION
**Pull request by brow**
This replaces #2101 with a new destination of `swift2`.

> This is an experiment with representing observers with functions Event<T, E> -> () rather than with SinkTypes. The experiment was suggested in #2053.

### CODE REVIEW

File: ReactiveCocoa/Swift/Event.swift [hunk 1]
```diff
@@ -27,11 +27,7 @@ public enum Event<T, E: ErrorType> {
 	/// will be received.
 	case Interrupted
 	
-	#if DEBUG
-	public typealias Sink = DebugSinkOf<Event>
-	#else
-	public typealias Sink = SinkOf<Event>
-	#endif
+	public typealias Sink = Event<T,E> -> ()
```
Comment (neilpa) [inline ReactiveCocoa/Swift/Event.swift]: space between type params please

Comment (brow) [inline ReactiveCocoa/Swift/Event.swift]: :white_check_mark:

**Comment by neilpa** [comment]
:100:

**Comment by jspahrsummers** [comment]
:sparkles:

**Pull request merged**